RecoTracker/FinalTrackSelectors/plugins/TrackCutClassifier.cc

```diff
@@ -48,61 +66,157 @@ namespace {
+      fillArrayI(minPixelHits, cfg,"minPixelHits");
+      fillArrayI(min3DLayers,  cfg,"min3DLayers");
+      fillArrayI(minLayers,    cfg,"minLayers");
       fillArrayI(maxLostLayers,cfg,"maxLostLayers");
-      fillArrayF(maxDz,cfg,"maxDz");
-      fillArrayF(maxDr,cfg,"maxDr");
-
+      fillArrayF(maxDz,        cfg,"maxDz");
+      fillArrayF(maxDr,        cfg,"maxDr");
+      edm::ParameterSet dz_par = cfg.getParameter<edm::ParameterSet>("dz_par");
+      fillArrayF(dz_par1,      dz_par,"dz_par1");
+      fillArrayF(dz_par2,      dz_par,"dz_par2");
+      edm::ParameterSet dr_par = cfg.getParameter<edm::ParameterSet>("dr_par");
+      fillArrayF(dr_par1,      dr_par,"dr_par1");
+      fillArrayF(dr_par2,      dr_par,"dr_par2");
     }
     
     
     
     float operator()(reco::Track const & trk,
-		   reco::BeamSpot const & beamSpot,
-		   reco::VertexCollection const & vertices,
-		   GBRForest const *) const {
+		     reco::BeamSpot const & beamSpot,
+		     reco::VertexCollection const & vertices,
+		     GBRForest const *) const {
 
-     
-      
       float ret = 1.f;
+      float dummy[3];
+      for (int i=2; i>=0; --i) dummy[i] = 1E-5;
+      ret = std::min(ret,cut(float(trk.ndof()),dummy,std::greater_equal<float>()) );
+
       auto  nLayers = trk.hitPattern().trackerLayersWithMeasurement();
       ret = std::min(ret,cut(nLayers,minLayers,std::greater_equal<int>()));
       if (ret==-1.f) return ret;
 
-      ret = std::min(ret,cut(chi2(trk)/float(nLayers),maxChi2n,std::less_equal<float>()));
+      ret = std::min(ret,cut(chi2n(trk)/float(nLayers),maxChi2n,std::less_equal<float>()));
       if (ret==-1.f) return ret;
-      ret = std::min(ret,cut(chi2(trk),maxChi2,std::less_equal<float>()));
+
+      ret = std::min(ret,cut(chi2n_no1Dmod(trk)/float(nLayers),maxChi2n_no1D,std::less_equal<float>()));
+      if (ret==-1.f) return ret;
+
+      ret = std::min(ret,cut(chi2n_no1Dmod(trk),maxChi2,std::less_equal<float>()));
       if (ret==-1.f) return ret;
      
       ret = std::min(ret,cut(n3DLayers(trk),min3DLayers,std::greater_equal<int>()));
       if (ret==-1.f) return ret;
+
       ret = std::min(ret,cut(nPixelHits(trk),minPixelHits,std::greater_equal<int>()));
       if (ret==-1.f) return ret;
+
       ret = std::min(ret,cut(lostLayers(trk),maxLostLayers,std::less_equal<int>()));
-     
+      if (ret==-1.f) return ret;
+      
+      // original dz and dr cut
       if (maxDz[2]<std::numeric_limits<float>::max() || maxDr[2]<std::numeric_limits<float>::max()) {
-        if (ret==-1.f) return ret;
-        Point bestVertex = getBestVertex(trk,vertices);
-        ret = std::min(ret,cut(dz(trk,bestVertex), maxDz,std::less_equal<float>()));
-        ret = std::min(ret,cut(dr(trk,bestVertex), maxDr,std::less_equal<float>()));
+
+	// if not primaryVertices are reconstructed, check compatibility w.r.t. beam spot
+        Point bestVertex = getBestVertex(trk,vertices); // min number of tracks 3
```

Comment:
I choose "2" in the getBestVertex.h because I thought it is fine for the offline
while I choose "3" for HLT, for back compatibility
opsi, thanks Matti for pointing that I missed to add the parameter !